hack/common.sh

```diff
@@ -103,10 +103,23 @@ sti::build::build_binaries() {
     for platform in "${platforms[@]}"; do
       sti::build::set_platform_envs "${platform}"
       echo "++ Building go targets for ${platform}:" "${targets[@]}"
-      go install "${goflags[@]:+${goflags[@]}}" \
-          -ldflags "${version_ldflags}" \
-          "${binaries[@]}"
-      sti::build::unset_platform_envs "${platform}"
+      # have to enable these debug flags for using gdb separately from goflags
+      # because the -gcflags "-N -l" ... i.e. the -N and -l are the setting for -gcflags
+      # not for go itself .. for now we punt on other goflags if developers are
+      # compiling to run with gdb
+      if [[ -v STI_ENABLE_GDB ]]; then
+	  echo "Enabling compiling go with debug to allow gdb usage (i.e. -gcflags \"-N -l\") since STI_ENABLE_GDB is set"
+	  echo "In this case, STI_ENABLE_GDB is set to  ${STI_ENABLE_GDB}"
+	  go install -gcflags "-N -l"\
+             -ldflags "${version_ldflags}" \
+             "${binaries[@]}"
+	  sti::build::unset_platform_envs "${platform}"
+      else
+	  go install "${goflags[@]:+${goflags[@]}}" \
+             -ldflags "${version_ldflags}" \
+             "${binaries[@]}"
+	  sti::build::unset_platform_envs "${platform}"
+      fi
```

Comment:
i'm still a little confused.  ignoring the actual script and STI_FLAGS env variable, is this not valid:
go install -gcflags "-N -1" -fooflag -barflag
?

I assume the arg to -gcflags needs to be quoted but given that, can't you set:
STI_FLAGS="-gcflags \"-N -1\""
?

i don't want to hold this PR up on this item, so if for now the simplest thing is "enabling debug means you don't get any other flags" i think we can live with it, but i am curious exactly what the issue is.  Probably something a 30 second conversation tomorrow can clarify.